hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java

```diff
@@ -101,6 +103,7 @@
 public final class BackupSystemTable implements Closeable {
 
   private static final Logger LOG = LoggerFactory.getLogger(BackupSystemTable.class);
+  private static final int BATCH_SIZE = 1000;
```

Comment:
Are all these `Put`s/`Delete`s against the same rowkey? If yes, you're implicitly getting an atomic row update via the current code path, I think.

If not, or you don't need row-level atomicity, you can replace this with a loop over calls to `BufferedMutator#mutate()`, and the that class will batch them up for RPC according to its configuration. Thus, it's still user-configurable, but using nobs that already exist. Unless you think that this particular operation needs some buffer settings that are specialized over the defaults.

Reply:
+1 on BufferedMutator.